pytorch_lightning/plugins/training_type/ddp_spawn.py

```diff
@@ -388,3 +388,13 @@ def register_plugins(cls, plugin_registry: Dict) -> None:
             description="DDPSpawn Plugin with `find_unused_parameters` as False",
             find_unused_parameters=False,
         )
+
+    def teardown(self) -> None:
+        if isinstance(self.model, DistributedDataParallel):
+            self.model = self.lightning_module
+
+        if self.on_gpu:
+            # GPU teardown
+            self.lightning_module.cpu()
+            # clean up memory
+            torch.cuda.empty_cache()
```

Comment:
Can we keep this part in the parallel plugin and call the super classes teardown here?
This is not specific to DDP but to every module basically.

Reply:
+1

Reply:
Yeah, the 396-400 still exist in parallel plugin. Added override teardown function in ddp plugins to unwrap the DistributedDataParallel model and get the lightning module first. I read through the comments history (@ananthsub @tchaton), seems that without unwrap there are gpu memory to held in ddp reducer? 
What's your recommendation? Will it be better to move the unwrap logic into other ddp functions and keep the teardown logic only in the super class (parallel)?

Reply:
@justusschock @awaelchli - i prefer keeping the parallel plugin as generic as possible as an interface. in the parent TrainingTypePlugin, `teardown` is an abstract method. 

the reason being, model parallel use cases have to extend the parallel plugin right now. for these use cases which have parts of the model on cpu and gpu, automatically moving the full lightning module to cpu at the end of training may not be desirable. keeping `teardown` as an abstract method here enforces that users should provide an implementation here

Reply:
@ananthsub re1: Sounds okay to me, but might be easier to forget it that way.

re2: Why wouldn't you want to completely move it to cpu?